### PR TITLE
Open food date

### PR DESCRIPTION
**Pull request by leezahuor**
# Issue/Task:
Implement `open` method for Food. This method should update the food item's expiration date to its used by date once it's opened. 

# Checklist
- [x] Fixed Issue/Accomplished Task
- [x] Wrote tests
- [x] All test cases passing

### CODE REVIEW

File: src/Food/food.py [hunk 1]
```diff
@@ -1,8 +1,16 @@
+from datetime import date, timedelta
+
 class Food(object):
     def __init__(self, name, expiration_date, use_by_length):
         self.name = name
-        self.expiration_date = expiration_date
-        self.use_by_length = use_by_length
+        self.expiration_date = date.fromisoformat(expiration_date)
+        self.use_by_length = timedelta(days = use_by_length)
```
Comment (amandate) [inline src/Food/food.py:7]: I feel like we should add a comment or change the name of `use_by_length` to make it more clear what it is

File: src/Food/food.py [hunk 2]
```diff
@@ -1,8 +1,16 @@
...
+        self.isOpen = False
     
-    def openItem():
-        pass 
\ No newline at end of file
+    '''Updates self.expiration_date to self.use_by_length according to date food item is opened.
```
Comment (amandate) [inline src/Food/food.py:10]: I think this comment should be adjusted slightly. We're updating `self.expiration_date` to the current date plus `self.use_by_length` and updating self.isOpen. I think saying "If date is changed, self.isOpen updates to True." Isn't exactly accurate basically.

File: tests/Food /test_food.py [hunk 3]
```diff
@@ -1,14 +1,24 @@
...
-        self.assertEqual("apple", food.name)
-        self.assertEqual("2022-5-4", food.expiration_date)
-        self.assertEqual(2, food.use_by_length)
+        self.assertEqual("apple", self.food.name)
+        self.assertEqual(date(2022, 5, 4), self.food.expiration_date)
+        self.assertEqual(timedelta(days = 2), self.food.use_by_length)
+        self.assertFalse(self.food.isOpen)  
+    
+    def testOpenFoods(self): 
+        self.assertFalse(self.food.isOpen)
```
Comment (amandate) [inline tests/Food /test_food.py:17]: Maybe add a comment saying you're checking the initial value of `isOpen`

**Review by amandate — COMMENTED** [review]
Just a few things!

File: src/Food/food.py (hunk 1 above)
Comment (leezahuor) [reply src/Food/food.py:7]: Done!

Comment (leezahuor) [reply src/Food/food.py]: Done!

File: src/Food/food.py (hunk 2 above)
Comment (leezahuor) [reply src/Food/food.py]: Done!

File: src/Food/food.py [hunk 4]
```diff
@@ -1,8 +1,17 @@
...
+
 class Food(object):
-    def __init__(self, name, expiration_date, use_by_length):
+    def __init__(self, name, expiration_date, use_by_date):
         self.name = name
-        self.expiration_date = expiration_date
-        self.use_by_length = use_by_length
+        self.expiration_date = date.fromisoformat(expiration_date)
+        
+        # use_by_date indicates the amount of time a food's quality will deteriorate after it is opened and suggests the food item should be consumed in that given timeframe.
```
Comment (amandate) [inline src/Food/food.py:8]:
```suggestion
        # use_by_date indicates the amount of time (in days) a food's quality will deteriorate after it is opened
```

**Review by amandate — APPROVED** [review]

**Pull request merged**